spec/pry_repl_spec.rb

```diff
@@ -121,4 +121,19 @@
       $stdout = old_stdout
     end
   end
+
+  describe "autoindent" do
+    it "should raise no exception when indented with a tab" do
+      ReplTester.start(correct_indent: true, auto_indent: true) do
+        output=@pry.config.output
+        def output.tty?; true; end
```

Comment:
Can we mock using RSpec?
If I am not mistaken, this test modifies the outside environment, which we'd want to prevent.

Reply:
> Can we mock using RSpec?

We could, but there is no reason to, since we don't want to test `tty?` here.

> If I am not mistaken, this test modifies the outside environment, which we'd want to prevent.

No, ReplTester.start invokes `redirect_pry_io input, output` where `output` is (essentially) a StringIO.new. So this is a temp object that only lives during the duration of the ReplTester block, and I am only adding to its singleton class, so this won't change the outside environment.

Reply:
I don't necessarily agree with this reasoning but I have to admit it's not a big deal.